application-mocca-calendar-ui/src/main/resources/MoccaCalendar/Macro.xml

```diff
@@ -489,9 +477,16 @@ if (typeof(XWiki) == "undefined" || typeof(XWiki.widgets) == "undefined" || type
               if (oldSkin) {
                 rightGroup.insert(this.createButton("button","$escapetool.javascript($services.localization.render('delete'))","","delete-moccacalendar-event","btn"));
               } else {
-                rightGroup.insert(' &lt;button class="button remove-button btn btn-danger" id="delete-moccacalendar-event"&gt; &lt;span class="glyphicon glyphicon-remove"&gt; &lt;/span&gt; '+"$escapetool.javascript($services.localization.render('delete'))"+'&lt;/button&gt;');
+                var button = ' &lt;button class="button remove-button btn btn-danger" type="button"' +
+                  'id="delete-moccacalendar-event" data-toggle="modal" data-target="#deleteEvents"&gt;'
+                var span = '&lt;span class="glyphicon glyphicon-remove"&gt; &lt;/span&gt; '
+                rightGroup.insert(button + span +
+                  "$escapetool.javascript($services.localization.render('delete'))"+'&lt;/button&gt;');
               }
-              rightGroup.down("#delete-moccacalendar-event").observe('click', this.showDeleteForm.bind(this));
+              var that = this;
+              jQuery(rightGroup.down('#delete-moccacalendar-event')).click(function() {
+                jQuery(this).data('calendarPopup', that);
```

Comment:
Why do you delay setting the data on click instead of setting it right away? Also, I wouldn't mix jQuery and Prototype.js on the same line:

    jQuery(rightGroup).find('#delete-moccacalendar-event').data('calendarPopup', this);